File: src/cross_tissue_fibrosis/paths.py
### environment

import os

### functions

def create_scvi_mod_ref(ref_mod_dir):
    return os.path.join(ref_mod_dir, "scvi")


def create_scvi_mod_query(q_mod_dir, ref_name):
    return os.path.join(q_mod_dir, "scvi_" + ref_name)


def create_scanvi_mod_ref(ref_mod_dir, meta):
    return os.path.join(ref_mod_dir, "scanvi_" + meta)


def create_scanvi_mod_query(q_mod_dir, ref_name, meta):
    return os.path.join(q_mod_dir, "scanvi_" + ref_name + "_" + meta)


def create_obj_file(obj_dir):
    return os.path.join(obj_dir, "adata.h5ad")


def create_labels_file(q_obj_dir, ref_name, meta):
    q_lab_hard = os.path.join(q_obj_dir, "scanvi_" + ref_name + "_" + meta + "_hard_labels.tsv")
    q_lab_soft = os.path.join(q_obj_dir, "scanvi_" + ref_name + "_" + meta + "_soft_labels.tsv")
    return q_lab_hard, q_lab_soft


def create_mod_path(parent_dir, name, batch_key, n_top_genes = 5000, flavor = "seurat_v3"):       
    dirname = os.path.join(parent_dir, "models", name, batch_key, flavor + "_" + str(n_top_genes))
    if not os.path.isdir(dirname):
        os.makedirs(dirname)    
    return dirname


def create_obj_path(parent_dir, name, batch_key, n_top_genes = 5000, flavor = "seurat_v3"):    
    dirname = os.path.join(parent_dir, "adata", name, batch_key, flavor + "_" + str(n_top_genes))
    if not os.path.isdir(dirname):
        os.makedirs(dirname)  
    return dirname


def create_milo_path(parent_dir, name, batch_key, n_top_genes = 5000, flavor = "seurat_v3"):
    dirname = os.path.join(parent_dir, "differential_analysis", name, batch_key, flavor + "_" + str(n_top_genes))
    if not os.path.isdir(dirname):
        os.makedirs(dirname)  
    return dirname

File: src/cross_tissue_fibrosis/paths.R

create_obj_path <- function(parent_dir, name, batch_key, n_top_genes, flavor) {
    dirname <- file.path(parent_dir, "adata", name, batch_key, paste0(flavor, "_", n_top_genes))
    dir.create(dirname, recursive = TRUE, showWarnings = FALSE) 
    return(dirname)
}


create_obj_file <- function(obj_dir) {
    return(file.path(obj_dir, "adata.h5ad"))
}


create_milo_path <- function(parent_dir, name, batch_key, n_top_genes = 5000, flavor = "seurat_v3") {
    dirname <- file.path(parent_dir, "differential_analysis", name, batch_key, paste0(flavor, "_", n_top_genes))
    dir.create(dirname, recursive = TRUE, showWarnings = FALSE) 
    return(dirname)
}


create_scvi_mod_query <- function(q_mod_dir, ref_name) {
    return(file.path(q_mod_dir, paste0("scvi_", ref_name)))
}


create_scanvi_mod_query <- function(q_mod_dir, ref_name, meta) {
    return(file.path(q_mod_dir, paste0("scanvi_", ref_name, "_", meta)))
}


create_labels_file <- function(q_obj_dir, ref_name, meta) {
    q_lab_hard <- file.path(q_obj_dir, paste0("scanvi_", ref_name, "_", meta, "_hard_labels.tsv"))
    q_lab_soft <- file.path(q_obj_dir, paste0("scanvi_", ref_name, "_", meta, "_soft_labels.tsv"))
    return(c(q_lab_hard, q_lab_soft))
}


